codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectEncoder.java

```diff
@@ -92,12 +92,16 @@ protected void encode(ChannelHandlerContext ctx, Object msg, List<Object> out) t
             buf = ctx.alloc().buffer((int) headersEncodedSizeAccumulator);
             // Encode the message.
             encodeInitialLine(buf, m);
-            encodeHeaders(m.headers(), buf);
-            headersEncodedSizeAccumulator = HEADERS_WEIGHT_NEW * padSizeForAccumulation(buf.readableBytes()) +
-                                            HEADERS_WEIGHT_HISTORICAL * headersEncodedSizeAccumulator;
-            ByteBufUtil.writeShortBE(buf, CRLF_SHORT);
             state = isContentAlwaysEmpty(m) ? ST_CONTENT_ALWAYS_EMPTY :
                     HttpUtil.isTransferEncodingChunked(m) ? ST_CONTENT_CHUNK : ST_CONTENT_NON_CHUNK;
+
+            filterMessage(m, state == ST_CONTENT_ALWAYS_EMPTY);
```

Comment:
consider renaming the method to clarify its purpose (e.g. `filterMessage` -> `sanitizeHeadersBeforeEncode`)

Reply:
done